src/plugins/output_format/common.h

```diff
@@ -139,103 +140,104 @@ namespace fmt
 {
 
 template<class T>
-struct __val_holder
+struct ValHolder
 {
     T value;
-
-    __val_holder(T&& v): value(std::forward<T>(v)) {}
+    ValHolder(T v): value(std::move(v)) {}
 };
 
 /* numeric value */
-template<class T>
-struct Nval: __val_holder<T>
+template<class T, class = void>
+struct Nval
 {
-    Nval(T&& v): __val_holder<T>(std::forward<T>(v)) {}
+    Nval(T v)
+    {
+        static_assert(always_false<T>::value, "should be integral type");
+    }
 };
 
-template<class T, std::enable_if_t<std::is_integral_v<std::remove_reference_t<T>>, int> = 0>
-auto nval(T&& value)
+template<class T>
+struct Nval<T, std::enable_if_t<std::is_integral_v<std::remove_reference_t<T>>, void>>: ValHolder<T>
 {
-    return Nval<T>(std::forward<T>(value));
-}
+    Nval(T v): ValHolder<T>(std::move(v)) {}
+};
 
 /* format specific numeric value */
 template<class T>
-struct Xval: __val_holder<T>
+struct Xval: Nval<T>
 {
     bool withbase;
-    Xval(T&& v, bool use_base = true): __val_holder<T>(std::forward<T>(v)), withbase(use_base) {}
+    Xval(T v, bool use_base = true): Nval<T>(std::move(v)), withbase(use_base) {}
```

Comment:
Isn't `std::forward` better than `std::move` here, because you're handling numeric values (which is probably trivial type) and forcing a move degrades performance?

Reply:
It's not a rvalue reference any more. ValHolder holds value and not tries to hold reference, so std:forward is useless.

Reply:
Oh, I derped.